### PR TITLE
[REVIEW] Add cuDF internals documentation (ColumnAccessor) 

### PR DESCRIPTION
**Pull request by shwina**
Begin documenting the internals of cuDF with `ColumnAccessor`. Also renames the `get_by*` methods of `ColumnAccessor` to `select_by*`.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: docs/cudf/source/internals.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+cuDF internals
+==============
+
+ColumnAccessor
+--------------
+
+The underlying data structure of cuDF `Series`, `DataFrame` and `Index` objects
+is a `ColumnAccessor`, which can be accessed via their `._data` attribute:
```
Comment (kkraus14) [inline docs/cudf/source/internals.md:8]: Should indicate that `ColumnAccessor` is underneath `Frame` which those classes inherit from.

File: docs/cudf/source/internals.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+```python
+>>> a = cudf.DataFrame({'x': [1, 2, 3], 'y': ['a', 'b', 'c']})
+>>> a._data
+ColumnAccessor(OrderedColumnDict([('x', <cudf.core.column.numerical.NumericalColumn object at 0x7f5a7d12e050>), ('y', <cudf.core.column.string.StringColumn object at 0x7f5a7d12e320>)]), multiindex=False, level_names=(None,))
+```
+
+`ColumnAccessor` is a dict-like object that maps column labels to columns.
+In addition, it supports things like selecting multiple columns (both by index and label),
+as well as hierarchical indexing.
```
Comment (kkraus14) [inline docs/cudf/source/internals.md:18]: Would clarify that `ColumnAccessor` is ordered, as dict-like isn't indicative of that.

**Pull request merged**